rustls/src/crypto/ring/tls12.rs

```diff
@@ -266,7 +266,7 @@ impl MessageEncrypter for GcmMessageEncrypter {
         let nonce = aead::Nonce::assume_unique_for_key(Nonce::new(&self.iv, seq).0);
         let aad = aead::Aad::from(make_tls12_aad(seq, msg.typ, msg.version, msg.payload.len()));
 
-        let total_len = msg.payload.len() + self.enc_key.algorithm().tag_len();
+        let total_len = self.encrypted_payload_len(msg.payload.len());
         let mut payload = Vec::with_capacity(GCM_EXPLICIT_NONCE_LEN + total_len);
```

Comment:
Shouldn't this be changed to `let mut payload = Vec::with_capacity(total_len)`? The `GCM_EXPLICIT_NONCE_LEN` is now accounted for in `encrypted_payload_len` right? I think this version adds the explicit nonce len twice (or I'm missing something)

Reply:
good catch